### DISCUSSION TITLE
[Proposal] Property Selection Lambdas with Pattern-Matching-Like Syntax

### DISCUSSION THREAD

**Discussion by ymarcov** [dotnet/csharplang#5352] [Language Ideas]
We often encounter situations where we need to get a particular property of an item:

```csharp
ICollection<Person> persons = GetPersons();
persons.Select(person => person.FirstName); // a frequent pattern which could benefit from a shorthand form
```

I propose to have a particular class of lambdas that only retrieve a property of their (single) argument. They can then have a shortened form:

```csharp
persons.Select({ FirstName }); // syntax sugar for .Select(person => person.FirstName)
```

In addition, it may be possible to have an extended class of lambdas that retrieve 2 or more properties as a tuple:

Thus the syntax sugar would work something like this:

```csharp
persons.Select({ FirstName }); // translates to .Select(person => (person.FirstName))
persons.Select({ FirstName, LastName }); // translates to .Select(person => (person.FirstName, person.LastName))
```

Regards,
Y.

**Comment by ymarcov** [comment]
Incidentally, this would save time in discussions regarding the naming conventions of such immediate lambda parameters.

**Reply by imakhlin** [reply]
In addition you may want to pass a typed object (<T> or IOptions<T>) and feel all matching fields within values from the <Person>

**Comment by HaloFour** [comment]
Dupe of #91

**Reply by ymarcov** [reply]
Thanks for that reference. I think there is an important difference in the proposed solution to this problem here, as I'd say it is a natural step forward from existing pattern matching syntax.

**Reply by ymarcov** [reply]
@HaloFour As an aside, would you happen to know if there is an open issue for method parameter unpacking from a tuple? Something like:
```csharp
var args = (1, 2, 3);
Sum(params args); // translates to Sum(1, 2, 3)
```

**Reply by HaloFour** [reply]
Yep, here it is: #424
